x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/delayeddatacheck/DelayedDataDetectorFactory.java

```diff
@@ -37,19 +38,21 @@ public static DelayedDataDetector buildDetector(Job job, DatafeedConfig datafeed
     }
 
     private static long validateAndCalculateWindowLength(TimeValue bucketSpan, TimeValue currentWindow) {
-        if (bucketSpan == null || currentWindow == null) {
+        if (bucketSpan == null) {
             return 0;
         }
-        if (currentWindow.compareTo(bucketSpan) < 0) {
-            // If it is the default value, we assume the user did not set it and the bucket span is very large
-            if (currentWindow.equals(DelayedDataCheckConfig.DEFAULT_DELAYED_DATA_WINDOW)) {
-                //TODO What if bucket_span > 24h?, weird but possible case...
-                return bucketSpan.millis() * FALLBACK_NUMBER_OF_BUCKETS_TO_SPAN;
+        if (currentWindow == null) { // we should provide a good default
+            if(bucketSpan.compareTo(DEFAULT_CHECK_WINDOW) >= 0) {
+                return FALLBACK_NUMBER_OF_BUCKETS_TO_SPAN * bucketSpan.millis();
             } else {
-                throw new IllegalArgumentException(
-                    Messages.getMessage(Messages.DATAFEED_CONFIG_DELAYED_DATA_CHECK_TOO_SMALL, currentWindow.getStringRep(),
-                        bucketSpan.getStringRep()));
+                return DEFAULT_CHECK_WINDOW.millis();
             }
+        }
+        if (currentWindow.compareTo(bucketSpan) < 0) {
+            // If it is the default value, we assume the user did not set it and the bucket span is very large
```

Comment:
This could no longer be the default value as `currentWindow` is not null, right?

Reply:
HAHA! Hey man! Everybody knows you shouldn't read the comments :D 

I will remove it :)